### PR TITLE
Introduce more optional chaining in the `src/core/` folder

### PR DESCRIPTION
**Pull request by Snuffleupagus**
After PR #12563 we're now free to use optional chaining in the worker-thread as well. (This patch also fixes one previously "missed" case in the `web/` folder.)

For the MOZCENTRAL build-target this patch reduces the total bundle-size by `1.6` kilobytes.

### CODE REVIEW

File: src/core/annotation.js [hunk 1]
```diff
@@ -2763,8 +2747,8 @@ class ButtonWidgetAnnotation extends WidgetAnnotation {
       return null;
     }
     const storageEntry = annotationStorage.get(this.data.id);
```
Comment (timvandermeij) [inline src/core/annotation.js:2749]: Shouldn't this line also use `annotationStorage?.get(this.data.id)` like the other places in the patch?

File: src/core/annotation.js [hunk 2]
```diff
@@ -2825,8 +2809,8 @@ class ButtonWidgetAnnotation extends WidgetAnnotation {
...
```
Comment (timvandermeij) [inline src/core/annotation.js:2811]: Same here, this likely also requires a `?` addition.

**Review by timvandermeij — APPROVED** [review]
r=me with the comments below addressed; thanks! (The comments below are potentially pre-existing issues, but given that similar code is touched in this PR I figured I'd mention them.)

File: src/core/annotation.js (hunk 1 above)
Comment (Snuffleupagus) [reply src/core/annotation.js:2749]: Just above we have an early return for `!annotationStorage`, hence that shouldn't be necessary in this case.

File: src/core/annotation.js (hunk 2 above)
Comment (Snuffleupagus) [reply src/core/annotation.js:2811]: Just above we have an early return for `!annotationStorage`, hence that shouldn't be necessary in this case.

File: src/core/annotation.js (hunk 1 above)
Comment (timvandermeij) [reply src/core/annotation.js:2749]: D'oh, I'm not sure how I missed that, but in that case all is fine!

**Pull request merged**